### DISCUSSION TITLE
Workflow triggered on push event does not take workflow from main branch

### DISCUSSION THREAD

**Discussion by Jonas1312** [community/community#62924] [Actions]
### Select Topic Area

Question

### Body

Hi,

I have this problem which is not a bug but rather a default behavior that I would like to change.

Let's say we have a workflow triggered with a push event on certain branches:

```yaml
on:
  push:
    branches:
      - main
      - "release/**"
```

And we also have this git history:

![image](https://github.com/community/community/assets/15847892/e206d167-3018-46f2-a370-640b57a792ca)

where the workflow on the main branch has been modified and is the most up-to-date.

We thus have two different workflows with the same name on the two branches.

If I push a commit on the release/0.1.0 branch, the workflow ran will be the old one from the release/0.1.0 branch, which is not the most up-to-date.

Is there a way to change this behavior, that is to always take the workflow from the main branch?

Thanks!

**Comment by mehulrajdev** [comment]
Clone the code first and use vs code. It will help.
